### ISSUE TITLE
Style Theme Windows do not work properly

### ISSUE THREAD

**Issue by RABINGG** [HeidiSQL/HeidiSQL#1974]
**Preconditions**
* HeidiSQL version: 12.7.0.6907
* Database type and version: MySql Version 8
* OS: Windows 11

**Describe the bug**
The style theme Windows should correspond with the OS theme. However, even when the theme is switched from light to dark in the OS, the theme does not change in HeidiSQL.

**To Reproduce**
1. Select style theme as windows in preferences.
2. Switch the theme of windows from light to dark in windows settings

**Expected**
The theme of HeidiSQL should be changed to dark to match the OS theme.

**Result**
The HeidiSQL doesn't change the theme

**Comment by ansgarbecker** [comment]
Although that's not what the "Windows" style is meant for, the user may expect that.
Also, there are multiple dark themes, so I will have to define one of them as the default.

**Comment by RABINGG** [comment]
> Although that's not what the "Windows" style is meant for, the user may expect that. Also, there are multiple dark themes, so I will have to define one of them as the default.

@ansgarbecker 
Any plans on implementing system adaptive theme?

**Comment by ansgarbecker** [comment]
System adaptive dark/light mode is now working in the new codebase. The v13 preview version from [the download page](https://www.heidisql.com/download.php#preview-windows-v13) can be tested with that.

Note that the various VCL themes like "Carbon" etc. are removed and unsupported in the new codebase. Instead, the controls either use adaptive dark/light mode, or the user forces light or dark mode in the preferences:

<img width="699" height="143" alt="Image" src="https://github.com/user-attachments/assets/72f2aac0-e9ee-48c2-aee9-d83ef2bb31b6" />

**Comment by RABINGG** [comment]
@ansgarbecker Still need a restart if the os theme switched, Is this expected?

**Comment by ansgarbecker** [comment]
Yes, this is one of the few option which need a restart. I recall I tested a live-switch but that showed some graphical glitches, so it seems better to restart once.
